Tests/ExtractFilesTests_Swift.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  ExtractDataTests.m
+//  UnzipKit
+//
+//  Created by Dov Frankel on 7/16/15.
+//  Copyright (c) 2015 Abbey Code. All rights reserved.
+//
+
+import UnzipKit
+
+class ExtractDataTestsSwift: UZKArchiveTestCase {
+
+    func testExtractData_NoPassword() {
+        let testArchives = ["Test Archive (Password).zip"]
+        for testArchiveName in testArchives {
+            do {
+                let archive = try UZKArchive(URL: self.testFileURLs[testArchiveName])
+                let data = try archive.extractDataFromFile("Test File A.txt", progress: { (percentDecompressed) -> Void in
+                    NSLog("Extracting, \(percentDecompressed) complete");
+                })
+            } catch let error as NSError {
```

Comment:
(This applies to both tests)

This is pretty close, but I think it would be better to declare a `var extractError: ErrorType!` outside of the `do/catch` and then assign it in the `catch` block. You can then remove the general catch, and assert that `extractError` (a) is not nil, (b) is an NSError, and (c) has the correct error code (as you were already asserting.

The problem with having your assertions inside the `catch` block is that that if there were no error thrown, or if it weren't an NSError, the test would still pass (since the assertions would be skipped over).

This is looking good though, thanks for sticking with it!